### PR TITLE
fix: password reset workflow

### PR DESCRIPTION
**Pull request by raastinn**

### CODE REVIEW

File: src/js/wall/Reset.jsx [hunk 1]
```diff
@@ -14,13 +14,22 @@ export class Reset extends React.Component {
...
         this.setState({ password: e.target.value });
     };
 
     handleSubmit = e => {
         e.preventDefault();
-        this.props.onReset(this.state.password, this.props.resetCode);
+
+        if (!this.state.password.length < 8) {
+            this.props.onReset(this.state.password, this.state.resetCode);
+        }
```
Comment (ReeceHoffmann) [inline src/js/wall/Reset.jsx]: Careful! This conditional always resolves to true as written
```suggestion
        if (!(this.state.password.length < 8)) {
            this.props.onReset(this.state.password, this.state.resetCode);
        }
```

Also: Will this render an error to the user if the request is blocked? I don't see it in the code, and I would hate to block the user but not say why

File: src/js/wall/Reset.jsx [hunk 2]
```diff
@@ -14,13 +14,22 @@ export class Reset extends React.Component {
         };
     }
 
+    componentDidMount() {
+        this.setState({
+            resetCode: this.props.resetCode,
```
Comment (ReeceHoffmann) [inline src/js/wall/Reset.jsx]: Did you figure out why the reset code get forgotten after the first request? Not totally against this soln if it makes the most sense, just seems strange that the code persists for only one request

**Comment by raastinn** [comment]
The reason why `resetCode` was being set to `undefined` when reset_password failed was the reducer changing the reset code I believe.

**Review by ReeceHoffmann — APPROVED** [review]

**Pull request merged**